android-core/src/main/java/com/mparticle/modernization/MParticleCallback.kt

```diff
@@ -0,0 +1,16 @@
+package com.mparticle.modernization
+
+abstract class MParticleCallback<S, E> {
```

Comment:
Does this introduce new behavior for a callback to make callbacks across the SDK more consistent?  Or is it just turning it into an abstract class to make it easier to grok?

Reply:
Exactly, its a way of making callbacks more consistent across. Each callback will have either success (S) or error (E) type. In Android we will be introducing suspend functions (kind of promises) for async behavior, but since though this is the "new standard", callbacks have been there for ages and Java libraries cant use suspend functions and coroutines, we added support for a consistent way of using callbacks.